AElf.CrossChain.Core/Helpers/ICrossChainContractReader.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Kernel;
+using AElf.Kernel.Types;
+
+namespace AElf.CrossChain
+{
+    public interface ICrossChainContractReader
+    {
+        Task<MerklePath> GetTxRootMerklePathInParentChainAsync(ulong blockHeight);
+        Task<ParentChainBlockData> GetBoundParentChainBlockInfoAsync(ulong height);
+        Task<ulong> GetBoundParentChainHeightAsync(ulong localChainHeight);
+        Task<ulong> GetParentChainCurrentHeightAsync(int chainId, Hash previousBlockHash,
+            ulong preBlockHeight);
+        Task<ulong> GetSideChainCurrentHeightAsync(int chainId, int sideChainId, Hash previousBlockHash,
+            ulong preBlockHeight);
+
+        Task<int> GetParentChainIdAsync(int chainId, Hash previousBlockHash, ulong preBlockHeight);
+        Task<Dictionary<int, ulong>> GetSideChainIdAndHeightAsync(int chainId, Hash previousBlockHash,
+            ulong preBlockHeight);
+        Task<IndexedSideChainBlockDataResult> GetIndexedSideChainBlockInfoResultAsync(ulong height);
+        Task<Dictionary<int, ulong>> GetAllChainsIdAndHeightAsync(int chainId, Hash blockHash, ulong blockHeight);
+    }
+    
+    public class CrossChainContractReader : ICrossChainContractReader
+    {
+        private readonly ICrossChainReadOnlyTransactionExecutor _crossChainReadOnlyTransactionExecutor;
+
+        public CrossChainContractReader(ICrossChainReadOnlyTransactionExecutor crossChainReadOnlyTransactionExecutor)
+        {
+            _crossChainReadOnlyTransactionExecutor = crossChainReadOnlyTransactionExecutor;
+        }
+        
+        public Task<MerklePath> GetTxRootMerklePathInParentChainAsync(ulong blockHeight)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public Task<ParentChainBlockData> GetBoundParentChainBlockInfoAsync(ulong height)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public Task<ulong> GetBoundParentChainHeightAsync(ulong localChainHeight)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public async Task<ulong> GetParentChainCurrentHeightAsync(int chainId, Hash previousBlockHash, ulong preBlockHeight)
+        {
+            var crossChainContractMethodAddress = ContractHelpers.GetCrossChainContractAddress(chainId);
```

Comment:
ContractHelpers should be removed.

cross chain address should get by service and saved/cached in your contract state